front/js/cart.js

```diff
@@ -1,92 +1,324 @@
+
+    for (let modif = 0; modif < elementModif.length; modif++){
+        elementModif[modif].addEventListener("change" , (event) => {
+            event.preventDefault()
+
+            //Selection de l'element à modifier en fonction de son id ET sa couleur
+            let quantityModif = productLocalStorage[modif].productQuantity
+            let elementModifValue = elementModif[modif].valueAsNumber
+            
+            const resultFound = productLocalStorage.find((element) => element.elementModifValue !== quantityModif)
+
+            resultFound.productQuantity = elementModifValue
+            productLocalStorage[modif].productQuantity = resultFound.productQuantity
+
+            localStorage.setItem("product", JSON.stringify(productLocalStorage))
+        
+            // refresh rapide
+            location.reload()
+        })
+    }
+}
+modifyQtt()
+
+function deleteProduct() {
+    let btn_supprimer = document.querySelectorAll(".deleteItem")
+
+    for (let suppr = 0; suppr < btn_supprimer.length; suppr++){
+        btn_supprimer[suppr].addEventListener("click" , (event) => {
+            event.preventDefault()
+
+            //Selection de l'element à supprimer en fonction de son id ET sa couleur
+            let idDelete = productLocalStorage[suppr].productId
+            let colorDelete = productLocalStorage[suppr].productColor
+
+            productLocalStorage = productLocalStorage.filter(element => element.productId !== idDelete || element.productColor !== colorDelete)
+            
+            localStorage.setItem("product", JSON.stringify(productLocalStorage))
+
+            //Alerte produit supprimé et refresh
+            alert("Ce produit a bien été supprimé du panier")
+            location.reload()
+        })
+    }
+}
+deleteProduct()
+
+function getForm() {
+    // Ajout des Regex
+    let form = document.querySelector(".cart__order__form")
+
+    //Création des expressions régulières
+    let emailRegExp = new RegExp('^[a-zA-Z0-9.-_]+[@]{1}[a-zA-Z0-9.-_]+[.]{1}[a-z]{2,10}$')
+    let charRegExp = new RegExp("^[a-zA-Z ,.'-]+$")
+    let addressRegExp = new RegExp("^[0-9]{1,3}(?:(?:[,. ]){1}[-a-zA-Zàâäéèêëïîôöùûüç]+)+")
+
+    // Ecoute de la modification du prénom
+    form.firstName.addEventListener('change', function() {
+        validFirstName(this)
+    });
+
+    // Ecoute de la modification du nom
+    form.lastName.addEventListener('change', function() {
+        validLastName(this)
+    });
+
+    // Ecoute de la modification de l'adresse
+    form.address.addEventListener('change', function() {
+        validAddress(this)
+    });
+
+    // Ecoute de la modification de la ville
+    form.city.addEventListener('change', function() {
+        validCity(this)
+    });
+
+    // Ecoute de la modification de l'email
+    form.email.addEventListener('change', function() {
+        validEmail(this)
+    });
+
+    //validation du prénom
+    const validFirstName = function(inputFirstName) {
```

Comment:
functions don't need to be declared within other functions. put them outside of this big one.